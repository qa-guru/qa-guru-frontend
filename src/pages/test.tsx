import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
