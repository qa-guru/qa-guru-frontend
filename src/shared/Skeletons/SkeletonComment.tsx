import { Skeleton, Stack } from "@mui/material";

const SkeletonComment = () => {
  return (
    <Stack mt="20px" spacing={1}>
      <Stack spacing={1}>
        <Stack direction="row" alignItems="center" spacing={1}>
          <Skeleton variant="circular" width={40} height={40} />
          <Stack width="10%">
            <Skeleton variant="text" sx={{ fontSize: "0.7rem" }} />
            <Skeleton variant="text" sx={{ fontSize: "0.7rem" }} />
          </Stack>
        </Stack>
        <Skeleton variant="rounded" width="100%" height={60} />
      </Stack>
      <Stack spacing={1}>
        <Stack direction="row" alignItems="center" spacing={1}>
          <Skeleton variant="circular" width={40} height={40} />
          <Stack width="10%">
            <Skeleton variant="text" sx={{ fontSize: "0.7rem" }} />
            <Skeleton variant="text" sx={{ fontSize: "0.7rem" }} />
          </Stack>
        </Stack>
        <Skeleton variant="rounded" width="100%" height={60} />
      </Stack>
    </Stack>
  );
};

export default SkeletonComment;
