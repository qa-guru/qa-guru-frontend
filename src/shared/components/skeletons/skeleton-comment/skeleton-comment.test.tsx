import { MemoryRouter } from "react-router-dom";
import { render } from "test/utils-test";

import SkeletonComment from "./skeleton-comment";

describe("SkeletonComment", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <SkeletonComment />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
