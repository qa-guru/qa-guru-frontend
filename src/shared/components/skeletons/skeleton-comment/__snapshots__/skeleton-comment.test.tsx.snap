// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SkeletonComment > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-e8agqp-MuiStack-root"
  >
    <div
      class="MuiStack-root css-jfdv4h-MuiStack-root"
    >
      <div
        class="MuiStack-root css-kdc3n8-MuiStack-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse css-wkdwxb-MuiSkeleton-root"
          style="width: 40px; height: 40px;"
        />
        <div
          class="MuiStack-root css-1r50ci7-MuiStack-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-12j6v5j-MuiSkeleton-root"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-12j6v5j-MuiSkeleton-root"
          />
        </div>
      </div>
      <span
        class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse css-7uzo9y-MuiSkeleton-root"
        style="width: 100%; height: 60px;"
      />
    </div>
    <div
      class="MuiStack-root css-jfdv4h-MuiStack-root"
    >
      <div
        class="MuiStack-root css-kdc3n8-MuiStack-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse css-wkdwxb-MuiSkeleton-root"
          style="width: 40px; height: 40px;"
        />
        <div
          class="MuiStack-root css-1r50ci7-MuiStack-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-12j6v5j-MuiSkeleton-root"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-12j6v5j-MuiSkeleton-root"
          />
        </div>
      </div>
      <span
        class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse css-7uzo9y-MuiSkeleton-root"
        style="width: 100%; height: 60px;"
      />
    </div>
  </div>
</DocumentFragment>
`;
